src/LanguageServer/Impl/Implementation/Server.WorkspaceSymbols.cs

```diff
@@ -16,56 +16,95 @@
+            // foreach (var entry in ProjectFiles) {
+            //     members = members.Concat(
+            //         await GetModuleVariablesAsync(entry as ProjectEntry, opts, @params.query, 50, cancellationToken)
+            //     );
+            // }
+            // 
+            // members = members.GroupBy(mr => mr.Name).Select(g => g.First());
+            // return members.Select(ToSymbolInformation).ToArray();
         }
 
         public override async Task<SymbolInformation[]> DocumentSymbol(DocumentSymbolParams @params, CancellationToken cancellationToken) {
-            await WaitForCompleteAnalysisAsync(cancellationToken);
-            var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly;
-            var entry = ProjectFiles.GetEntry(@params.textDocument.uri);
-
-            var members = await GetModuleVariablesAsync(entry as ProjectEntry, opts, string.Empty, 50, cancellationToken);
-            return members
-                .GroupBy(mr => mr.Name)
-                .Select(g => g.First())
-                .Select(ToSymbolInformation)
-                .ToArray();
+            // var uri = @params.textDocument.uri;
+            // var symbols = await SymbolIndex.HierarchicalDocumentSymbolsAsync(uri, cancellationToken);
+            // var flat = symbols.Flatten(uri);
+            
+            throw new Exception("this function is not called");
+
+            // var uri = @params.textDocument.uri;
+            // var symbols = await SymbolIndex.WorkspaceSymbolsAsync(string.Empty, cancellationToken);
+            // return Array.Empty<SymbolInformation>();
+
+            // var ast = GetCurrentAst(uri);
+            // var walker = new IndexerWalker(ast);
+            // ast.Walk(walker);
+            // return IndexerWalker.DocSymsToSymInfos(uri, walker.Symbols).ToArray();
+            // 
+            // await WaitForCompleteAnalysisAsync(cancellationToken);
+            // var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly;
+            // var entry = ProjectFiles.GetEntry(@params.textDocument.uri);
+            // 
+            // var members = await GetModuleVariablesAsync(entry as ProjectEntry, opts, string.Empty, 50, cancellationToken);
+            // return members
+            //     .GroupBy(mr => mr.Name)
+            //     .Select(g => g.First())
+            //     .Select(ToSymbolInformation)
+            //     .ToArray();
         }
 
         public override async Task<DocumentSymbol[]> HierarchicalDocumentSymbol(DocumentSymbolParams @params, CancellationToken cancellationToken) {
-            await WaitForCompleteAnalysisAsync(cancellationToken);
-            var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly;
-            var entry = ProjectFiles.GetEntry(@params.textDocument.uri);
+            var uri = @params.textDocument.uri;
+            var symbols = await SymbolIndex.HierarchicalDocumentSymbolsAsync(uri, cancellationToken);
```

Comment:
This could then await on a `TaskCompletionSource` that would be set when indexing for this particular URI completes.

Reply:
Resolving, as this will come in a future PR.